### PR TITLE
LAWS-129 Utah calculation changes to the new federal 2020 form

### PR DESCRIPTION
**Pull request by Aidancs**
**QA Approved**
refs: https://spurjob.atlassian.net/browse/LAWS-14
refs: https://spurjob.atlassian.net/browse/LAWS-114

### CODE REVIEW

File: src/Countries/US/Utah/UtahIncome/V20190101/UtahIncome.php [hunk 1]
```diff
@@ -7,52 +7,9 @@
...
-    {
-        $annual_withholding_allowance = 0;
-        if ($this->tax_information->filing_status === 'S') {
-            $annual_withholding_allowance = self::SINGLE_BASE_ALLOWANCE - $this->getAnnualWithholdingAllowanceReduction();
-        } else {
-            $annual_withholding_allowance = self::MARRIED_BASE_ALLOWANCE - $this->getAnnualWithholdingAllowanceReduction();
-        }
-
-        return $annual_withholding_allowance > 0 ? $annual_withholding_allowance : 0;
+        return 0.0;
```
Comment (stevespur) [inline src/Countries/US/Utah/UtahIncome/V20190101/UtahIncome.php]: why are we changing 2019?

Comment (Aidancs) [reply src/Countries/US/Utah/UtahIncome/V20190101/UtahIncome.php]: No one has used it and I tried to delete it but the unemployment test referenced it

File: src/Countries/US/Utah/UtahIncome/V20200101/UtahIncome.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        [51270, 0.24, 6480],
+        [89540, 0.32, 15666],
+        [111230, 0.35, 22604],
+        [264320, 0.37, 76185],
+    ];
+
+    public function __construct(UtahIncomeTaxInformation $tax_information, Payroll $payroll)
+    {
+        parent::__construct($tax_information, $payroll);
+        $this->tax_information = $tax_information;
```
Comment (stevespur) [inline src/Countries/US/Utah/UtahIncome/V20200101/UtahIncome.php]: minor detail but doesn't the parent set this?

File: src/Countries/US/Utah/UtahIncome/V20200101/UtahIncome.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        if ($this->isUserClaimingExemption()) {
+            return 0.00;
+        }
+
+        $taxable_wages = $this->getAdjustedWageAmount();
+        $taxable_wages = $this->getTentativeAmount($taxable_wages);
+        $taxable_wages -= $this->tax_information->dependents_deduction_amount;
+        $taxable_wages = $taxable_wages > 0 ? $taxable_wages : 0;
+        $taxable_wages = $taxable_wages > 0 ? $taxable_wages / $this->payroll->pay_periods : 0;
```
Comment (stevespur) [inline src/Countries/US/Utah/UtahIncome/V20200101/UtahIncome.php]: feels like these should be combined

File: src/Countries/US/Utah/UtahIncome/V20200101/UtahIncome.php [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    public function compute(Collection $tax_areas)
+    {
+        if ($this->isUserClaimingExemption()) {
+            return 0.00;
+        }
+
+        $taxable_wages = $this->getAdjustedWageAmount();
+        $taxable_wages = $this->getTentativeAmount($taxable_wages);
```
Comment (stevespur) [inline src/Countries/US/Utah/UtahIncome/V20200101/UtahIncome.php]: variable name is a bit confusing as the return here is the tax amount not the taxable wages?

**Pull request merged**